src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -775,4 +786,114 @@ public void validateScoreSettings(Exercise exercise) {
+
+        List<Result> results = resultRepository.findWithEagerSubmissionAndFeedbackByParticipationExerciseId(exercise.getId());
+
+        // re-calculate the results after updating the feedback
+        for (Result result : results) {
+            if (!feedbackToBeDeleted.isEmpty()) {
+                List<Feedback> existingFeedback = result.getFeedbacks();
+                if (!existingFeedback.isEmpty()) {
+                    existingFeedback.removeAll(feedbackToBeDeleted);
+                }
+                // first save the feedback (that is not yet in the database) to prevent null index exception
+                List<Feedback> savedFeedback = feedbackRepository.saveFeedbacks(existingFeedback);
+                result.updateAllFeedbackItems(savedFeedback, exercise instanceof ProgrammingExercise);
+            }
+
+            if (!(exercise instanceof ProgrammingExercise)) {
+                resultRepository.submitResult(result, exercise);
+            }
+            else {
+                double totalScore = programmingAssessmentService.calculateTotalScore(result);
+                result.setScore(totalScore, exercise.getMaxPoints());
+                /*
+                 * Result string has following structure e.g: "1 of 13 passed, 2 issues, 10 of 100 points" The last part of the result string has to be updated, as the points the
+                 * student has achieved have changed
+                 */
+                String[] resultStringParts = result.getResultString().split(", ");
+                resultStringParts[resultStringParts.length - 1] = result.createResultString(totalScore, exercise.getMaxPoints());
+                result.setResultString(String.join(", ", resultStringParts));
+                resultRepository.save(result);
+            }
+        }
+    }
+
+    /**
+     * Gets the list of feedback that is associated with deleted grading instructions
+     *
+     * @param deleteFeedbackAfterGradingInstructionUpdate  boolean flag that indicates whether the associated feedback should be deleted or not
+     * @param gradingInstructions grading instruction list to update
+     * @param exercise exercise for which the grading instructions have to be updated
+     * @return list including Feedback entries that have to be deleted due to updated grading instructions
+     */
+    public List<Feedback> getFeedbackToBeDeletedAfterGradingInstructionUpdate(boolean deleteFeedbackAfterGradingInstructionUpdate, List<GradingInstruction> gradingInstructions,
+            Exercise exercise) {
+        List<Feedback> feedbackToBeDeleted = new ArrayList<>();
+        // check if the user decided to remove the feedback after deleting the associated grading instructions
+        if (deleteFeedbackAfterGradingInstructionUpdate) {
+            List<Long> updatedInstructionIds = gradingInstructions.stream().map(GradingInstruction::getId).collect(Collectors.toList());
```

Comment:
```suggestion
            List<Long> updatedInstructionIds = gradingInstructions.stream().map(GradingInstruction::getId).toList();
```
Same here with the list